tests/src/Integration/Condition/ListContainsTest.php

```diff
@@ -139,4 +139,30 @@ public function testConditionEvaluation() {
       ->setContextValue('item', $entity_four);
     $this->assertFalse($this->condition->evaluate());
   }
+
+  /**
+   * @covers ::refineContextDefinitions
+   */
+  public function testRefiningContextDefinitions() {
+    // Create array of mock entities
+    $entity_zero = $this->getMock('Drupal\Core\Entity\EntityInterface');
+    $entity_zero->expects($this->any())
+      ->method('id')
+      ->will($this->returnValue('entity_zero_id'));
+    
+    $entity_one = $this->getMock('Drupal\Core\Entity\EntityInterface');
+    $entity_one->expects($this->any())
+      ->method('id')
+      ->will($this->returnValue('entity_one_id'));
+
+    // Test array of entities
+    $entity_list = [$entity_zero, $entity_one];
+
+    $this->condition
+      ->setContextValue('list', $entity_list)
+      ->setContextValue('item', $entity_zero);
+    $this->condition->refineContextdefinitions();
+    $this->assertEquals(
+      $this->condition->getContextValue('item'), $entity_zero);
```

Comment:
Actually we should get the context definition here for the item and make sure that the type is now the entity type.